src/uct/ib/base/ib_pd.c

```diff
@@ -400,6 +406,24 @@ uct_ib_pd_open(const char *pd_name, const uct_pd_config_t *uct_pd_config, uct_pd
         goto err_free_pd;
     }
 
+    if (pd_config->fork_init != UCS_NO) {
+        ret = ibv_fork_init();
+        if (ret) {
+            if (pd_config->fork_init == UCS_YES) {
+                ucs_error("ibv_fork_init() failed: %m");
+                status = UCS_ERR_IO_ERROR;
+                goto err_release_stats;
+            }
+            if (errno != ENOENT) {
+                /* ENOENT returned when ibv_fork_init attempted after an
+                 * interoperable communication library already initialized
+                 * without doing ibv_fork_init, and created qpairs. Lets not
+                 * report this error to the end-user in this case. */
+                ucs_warn("ibv_fork_init() failed: %m");
```

Comment:
then i would suggest to remove this warning, since anyway it would not print a warning in the most common case of ibv_fork_init() failure.
@alinask what do you think?